R/uvhydrograph-render.R

```diff
@@ -45,11 +47,25 @@ createPrimaryPlot <- function(data, month){
     
     correctionLabels <- parseLabelSpacing(primaryData[i], primaryInfo)
     primaryApprovals <-  parseApprovalInfo(primaryData[i], primaryInfo, x, y)
-    primaryStyles <- getUvStyle(primaryData[i], primaryInfo, x, y, primaryApprovals, correctionLabels, "primary")
-    primaryPlotTypes <- getPlotType(primaryData[i], "primary")
-   
-    if (names(primaryData[i]) %in% c("series_corr", "meas_Q")) {
-      for (j in 1:length(primaryStyles)) {
+    
+    if (length(primaryApprovals) > 0) {
+      primaryStyles <- vector("list", length(primaryApprovals))
+      primaryPlotTypes <- c()
+      notPlot <- c()
+      for (k in seq(length(primaryApprovals))) {
+        primaryStyles[[k]] <- getUvStyle(primaryData[i], primaryInfo, x, y, primaryApprovals[[k]], correctionLabels, "primary")
+        primaryPlotTypes[k] <- getPlotType(primaryData[i], "primary")
+        notPlot[k] <- all(is.na(primaryStyles[[k]]$x))
+      }
+      primaryStyles[which(notPlot)] <- NULL
+      primaryPlotTypes <- primaryPlotTypes[which(!notPlot)]
+    } else {
+      primaryStyles <- getUvStyle(primaryData[i], primaryInfo, x, y, primaryApprovals, correctionLabels, "primary")
+      primaryPlotTypes <- getPlotType(primaryData[i], "primary")
+    }
+    
+    if (names(primaryData[i]) %in% c("series_corr", "meas_Q", "max_DV", "min_DV", "median_DV", "mean_DV", "UV_series")) {
+      for (j in seq(length(primaryStyles))) {
```

Comment:
I think you want `seq_len(length(primaryStyles))`, because seq(0) is `1 2` vs seq_len(0) is NULL (and won't enter the loop)